Automattic/jetpack
# Identity Crisis: Move the check for an idc to the identity-crisis package

#### Changes proposed in this Pull Request:
* Add a new method, `Identity_Crisis::check_response_for_idc`, which will check whether the response contains an IDC error code.
* Use this method in Sync to determine if the Sync request returned an error code.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions

Testing will consist of verifying that an IDC that's detected by a Sync request works as expected.
The 'idc_detected" key in the request body is not yet being sent by WPCOM.

1. Install this branch of Jetpack. Activate and connect to WPCOM.
2. Activate the Jetpack Debug plugin.
3. Enable the IDC simulator module in the Jetpack Debug menu.
4. Enable the IDC simulation in the UI.
5. Create a draft post,
6. Observe your site in IDC safe mode.
7. Check the site's debug.log file and verify that no errors were generated.
